api/src/services/assets.ts

```diff
@@ -178,7 +180,20 @@ export class AssetsService {
 				readStream.unpipe(transformer);
 			});
 
-			await storage.location(file.storage).write(assetFilename, readStream.pipe(transformer), type);
+			try {
+				await storage.location(file.storage).write(assetFilename, readStream.pipe(transformer), type);
+			} catch (error) {
+				await storage.location(file.storage).delete(assetFilename);
+
+				if ((error as Error)?.message?.includes('timeout')) {
+					throw new ServiceUnavailableError({
+						service: 'asset',
+						reason: `Transformation timed out after ${timeoutSeconds} seconds`,
```

Comment:
```suggestion
						reason: `Transformation timed out after ${timeoutSeconds}s`,
```

Super tiny thing, but copying the same approach from:

https://github.com/directus/directus/blob/523c57739cedc5795b3ea2c7f2c9471aef933988/api/src/websocket/controllers/base.ts#L189

that way `1s` can probably fit better compared to `1 seconds` 😄